scripts/generate_abi.py

```diff
@@ -0,0 +1,97 @@
+import json
+import tempfile
+from pathlib import Path
+from typing import Any
+
+import click
+from psutil import Popen
+
+COMPILE_COMMAND = """
+git clone git@github.com:beamer-bridge/beamer.git {beamer_path} &&
+cd {beamer_path} &&
+git fetch &&
+git checkout {commit_id} &&
+$(which ape) compile
+"""
+
+
+def _get_contract_commit_ids(deployment_path: Path) -> dict[str, set[str]]:
+    file_contract_commits: dict[str, set[str]] = {}
+    for file_path in deployment_path.glob("*.json"):
+        with open(file_path, "r") as file:
+            data = json.load(file)
+            data.pop("deployer")
+            for contract_name, contract_info in data.items():
+                if contract_name not in file_contract_commits:
+                    file_contract_commits[contract_name] = set()
+                file_contract_commits[contract_name].add(contract_info["beamer_commit"])
```

Comment:
Once you load the json from file, there is no need to keep the file open so we can just move these out of the context manager.